src/main/cpp/src/api/genomicsdb.cc

```diff
@@ -0,0 +1,539 @@
+                         const VariantQueryProcessor& query_processor,
+                         const VidMapper& vid_mapper) {
+  std::string indent_prefix = "    ";
+  std::cout << "{\n";
+  //variant_calls is an array of dictionaries
+  std::cout << indent_prefix << "\"variant_calls\": [\n";
+  VariantCallPrintOperator printer(std::cout, indent_prefix+indent_prefix, &vid_mapper);
+  query_processor.iterate_over_cells(query_processor.get_array_descriptor(), query_config, printer, true);
+  std::cout << "\n" << indent_prefix << "]\n";
+  std::cout << "}\n";
+}
+
+std::vector<VariantCall>* GenomicsDB::query_variant_calls(const std::string& array, VariantQueryConfig *query_config, GenomicsDBVariantCallProcessor& processor) {
+  #if(0)
+  auto query_timer = Timer();
+  query_timer.start();
+#endif
+
+  // Get a Variant Query Processor
+  VariantQueryProcessor *query_processor = new VariantQueryProcessor(
+      TO_VARIANT_STORAGE_MANAGER(m_storage_manager), array, query_config->get_vid_mapper());
+  query_processor->do_query_bookkeeping(query_processor->get_array_schema(),
+                                        *query_config, query_config->get_vid_mapper(), true);
+
+#if(0)
+  print_variant_calls(*query_config, *query_processor, query_config->get_vid_mapper());
+#endif
+
+  // Perform Query over all the intervals
+  std::vector<VariantCall> *pvariant_calls = new std::vector<VariantCall>;
+  VidMapper* vid_mapper;
+  if (m_vid_mapper)  {
+    vid_mapper = TO_VID_MAPPER(m_vid_mapper);
+  } else {
+    vid_mapper = const_cast<VidMapper *>(&query_config->get_vid_mapper());
+  }
+  GatherVariantCalls gather_variant_calls(processor, get_vid_mapper(m_vid_mapper, query_config));
+  query_processor->iterate_over_cells(query_processor->get_array_descriptor(), *query_config, gather_variant_calls, true);
+
+  delete query_processor;
+
+#if(0)
+  query_timer.stop();
+  query_timer.print();
+#endif
+
+  return pvariant_calls;
+}
+
+// Template to get the mapped interval from the GenomicsDB array for the Variant(Call)
+template<class VariantOrVariantCall>
+interval_t get_interval_for(const VariantOrVariantCall* variant_or_variant_call) {
+  return std::make_pair<uint64_t, uint64_t>(variant_or_variant_call->get_column_begin(), variant_or_variant_call->get_column_end());
+}
+
+interval_t GenomicsDB::get_interval(const genomicsdb_variant_t* variant) {
+  return get_interval_for(TO_VARIANT(variant));
+}
+
+interval_t GenomicsDB::get_interval(const genomicsdb_variant_call_t* variant_call) {
+  return get_interval_for(TO_VARIANT_CALL(variant_call));
+}
+
+// Template to get the genomic interval from the GenomicsDB array associated with the Variant(Call)
+template<class VariantOrVariantCall>
+genomic_interval_t get_genomic_interval_for(const VariantOrVariantCall* variant_or_variant_call, VidMapper *vid_mapper) {
+  std::string contig_name;
+  int64_t contig_position;
+  vid_mapper->get_contig_location(variant_or_variant_call->get_column_begin(), contig_name,
```

Comment:
we should check the return from vid_mapper->get_contig_location here to ensure that column successfully mapped to contig_name/_position